### ISSUE TITLE
Allow textual empty label as a property in fields and inputs

### ISSUE THREAD

**Issue by GuillermoBlasco** [marmelab/react-admin#2544]
Context: 

Whenever a field is empty, the display text is empty as well. See the example, where the field `Pitch` is a ReferenceField with allowedEmpty enabled. 
![image](https://user-images.githubusercontent.com/1889971/48705606-e370ba80-ebf9-11e8-8054-1373b9411b48.png)
Also whenever an input allows for empty value the empty value is displayed with an empty text:
![image](https://user-images.githubusercontent.com/1889971/48706043-43b42c00-ebfb-11e8-9bf5-20384aefda86.png)

Request:

It woule be nice to display an actual string that suggests that emptyness instead of just empty. It could be configured as a field and input property.

Field case:
![image](https://user-images.githubusercontent.com/1889971/48705946-f2a43800-ebfa-11e8-9f2d-fa4655745f0d.png)

Possible configuration:
```
        <ReferenceField linkType="show" label="Pitch" source="pitch" reference="pitches" allowEmpty emptyLabel="None">
          <TextField source="name" required />
        </ReferenceField>
```

Input case:
![image](https://user-images.githubusercontent.com/1889971/48706123-942b8980-ebfb-11e8-93ff-de911aef9b03.png)

Possible configuration:
```
      <ReferenceInput label="Pitch" source="pitch" reference="pitches" allowEmpty emptyLabel="None">
        <SelectInput optionText="name" />
      </ReferenceInput>
```

**Comment by fzaninotto** [comment]
Nice idea! Would you mind working on a PR to add this feature?

**Comment by fzaninotto** [comment]
Fixed by #4413
